lib/ruby-next/core/enumerable/tally.rb

```diff
@@ -1,12 +1,15 @@
 # frozen_string_literal: true
 
-# Refine Array seprately, 'cause refining modules is vulnerable to prepend:
+# Refine Array separately, 'cause refining modules is vulnerable to prepend:
 # - https://bugs.ruby-lang.org/issues/13446
-RubyNext::Core.patch Enumerable, method: :tally, version: "2.7", refineable: [Enumerable, Array] do
+RubyNext::Core.patch Enumerable, method: :tally, version: "3.1", supported: {}.respond_to?(:tally) && {}.method(:tally).arity < 0, refineable: [Enumerable, Array] do
```

Comment:
It means that we use our own version in Ruby 3.0 even without arguments; could you please make a quick benchmark to compare our Ruby implementation with the built-in Ruby `#tally` in case no Hash arg is provided?

I'd like to avoid a performance regression here.

Reply:
It was x2 slower, so now we call `super` when it's available.